exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/UnionFunctions.java

```diff
@@ -32,16 +33,72 @@
 import org.apache.drill.exec.expr.holders.UnionHolder;
 import org.apache.drill.exec.expr.holders.IntHolder;
 import org.apache.drill.exec.expr.holders.VarCharHolder;
+import org.apache.drill.exec.resolver.TypeCastRules;
 import org.apache.drill.exec.vector.complex.impl.UnionReader;
 import org.apache.drill.exec.vector.complex.reader.FieldReader;
 
 import javax.inject.Inject;
+import java.util.Set;
 
 /**
  * The class contains additional functions for union types in addition to those in GUnionFunctions
  */
 public class UnionFunctions {
 
+  /**
+   * Returns zero if the inputs have equivalent types. Two numeric types are considered equivalent, as are a combination
+   * of date/timestamp. If not equivalent, returns a value determined by the numeric value of the MinorType enum
```

Comment:
let's add an additional ordering ordinal to the minortype enum so that we match Mongo's ordering among types